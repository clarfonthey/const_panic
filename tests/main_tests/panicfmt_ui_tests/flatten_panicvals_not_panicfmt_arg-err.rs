use const_panic::{FmtArg, PanicVal, flatten_panicvals};

struct NotFmt;

const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; NotFmt);

const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; NotFmt => NotFmt);

fn main() {}