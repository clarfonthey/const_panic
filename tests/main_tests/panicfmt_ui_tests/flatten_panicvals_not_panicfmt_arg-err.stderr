error[E0277]: the trait bound `NotFmt: PanicFmt` is not satisfied
 --> tests/main_tests/panicfmt_ui_tests/flatten_panicvals_not_panicfmt_arg-err.rs:5:29
  |
5 | const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; NotFmt);
  |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `PanicFmt` is not implemented for `NotFmt`
  |
  = help: the following other types implement trait `PanicFmt`:
            &'a T
            ()
            *const T
            *mut T
            ArrayString<CAP>
            Delimiter
            FromBytesUntilNulError
            FromBytesWithNulError
          and $N others
  = note: this error originates in the macro `$crate::__to_pvf_kind` which comes from the expansion of the macro `flatten_panicvals` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotFmt: PanicFmt` is not satisfied
 --> tests/main_tests/panicfmt_ui_tests/flatten_panicvals_not_panicfmt_arg-err.rs:7:63
  |
7 | const _: &[PanicVal<'_>] = &flatten_panicvals!(FmtArg::DEBUG; NotFmt => NotFmt);
  |                                                               ^^^^^^ the trait `PanicFmt` is not implemented for `NotFmt`
  |
  = help: the following other types implement trait `PanicFmt`:
            &'a T
            ()
            *const T
            *mut T
            ArrayString<CAP>
            Delimiter
            FromBytesUntilNulError
            FromBytesWithNulError
          and $N others
